### ISSUE TITLE
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)

### ISSUE THREAD

**Issue by halogbi** [haccer/twint#1312]
# Issue CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)

I already install and upgrade 
I have python 3.7

it's not work never ever why . how can I fix this iiisue ASAP .

CRITICAL:root:twint.get:User:'NoneType' object is not subscriptable
CRITICAL:root:twint.run:Twint:Feed:noDataExpecting value: line 1 column 1 (char 0)
sleeping for 15 secs

**Comment by timothydillan** [comment]
Hey there. I got this issue as well and managed to fix it. All you need to do is reinstall twint, and upgrade it: 
`pip3 uninstall twint && pip3 install twint && pip3 install --user --upgrade git+https://github.com/twintproject/twint.git@origin/master#egg=twint`
<img width="1131" alt="Screen Shot 2021-12-24 at 10 11 16" src="https://user-images.githubusercontent.com/55008869/147307230-9c5b80c6-138c-41d9-9a6d-ce2ee5bbc76f.png">
Hope this helps!

**Comment by smthjhn** [comment]
> Hey there. I got this issue as well and managed to fix it. All you need to do is reinstall twint, and upgrade it: `pip3 uninstall twint && pip3 install twint && pip3 install --user --upgrade git+https://github.com/twintproject/twint.git@origin/master#egg=twint` <img alt="Screen Shot 2021-12-24 at 10 11 16" width="1131" src="https://user-images.githubusercontent.com/55008869/147307230-9c5b80c6-138c-41d9-9a6d-ce2ee5bbc76f.png"> Hope this helps!

What version of python are you using? I tried using python 3.6.15 but it still gave me the same error. I also tried python 3.10.1 but I got `<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject
[-] TWINT requires Python version 3.6+.`

UPDATE: python 3.9.9 worked for me

**Comment by timothydillan** [comment]
> > Hey there. I got this issue as well and managed to fix it. All you need to do is reinstall twint, and upgrade it: `pip3 uninstall twint && pip3 install twint && pip3 install --user --upgrade git+https://github.com/twintproject/twint.git@origin/master#egg=twint` <img alt="Screen Shot 2021-12-24 at 10 11 16" width="1131" src="https://user-images.githubusercontent.com/55008869/147307230-9c5b80c6-138c-41d9-9a6d-ce2ee5bbc76f.png"> Hope this helps!
> 
> What version of python are you using? I tried using python 3.6.15 but it still gave me the same error. I also tried python 3.10.1 but I got `<frozen importlib._bootstrap>:241: RuntimeWarning: greenlet.greenlet size changed, may indicate binary incompatibility. Expected 152 from C header, got 160 from PyObject [-] TWINT requires Python version 3.6+.`
> 
> UPDATE: python 3.9.9 worked for me

Yeah, you can't use 3.10 and above since there's a bug in the library that incorrectly detects your Python version. Check this PR out: https://github.com/twintproject/twint/pull/1309

**Comment by timothydillan** [comment]
Alternatively, if twint doesn't really work out for you, you can instead use snscrape https://medium.com/dataseries/how-to-scrape-millions-of-tweets-using-snscrape-195ee3594721. I find it less susceptible to "Guest token not found" errors and provides complete tweets.

**Comment by weirdyang** [comment]
> pip3 uninstall twint && pip3 install twint && pip3 install --user --upgrade

this didn't fix it for me, I'm using Python 3.9.1

any other advice?

**Comment by timothydillan** [comment]
> > pip3 uninstall twint && pip3 install twint && pip3 install --user --upgrade
> 
> this didn't fix it for me, I'm using Python 3.9.1
> 
> any other advice?

Alternatively you can use [snscrape](https://github.com/JustAnotherArchivist/snscrape). Here's a basic guide on using it: https://medium.com/dataseries/how-to-scrape-millions-of-tweets-using-snscrape-195ee3594721.

**Comment by weirdyang** [comment]
For those with this issue, this worked for me 

https://github.com/twintproject/twint/issues/915#issuecomment-813015561

Remember  to restart the kernel
